packages/expo-updates/ios/EXUpdates/Logging/UpdatesLogEntry.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import Foundation
+
+/**
+ Schema for the fields in expo-updates log message JSON strings
+ */
+public struct UpdatesLogEntry: Codable {
+  var timestamp: UInt // seconds since 1/1/1970 UTC
+  var message: String
+  var code: String // One of the UpdatesErrorCode string values
+  var level: String // One of the ExpoModulesCore.LogType string values
+  var updateId: String? // EAS update ID, if any
+  var assetId: String? // EAS asset ID, if any
+  var stacktrace: [String]? // Stacktrace (for error and fatal logs)
+
+  /**
+   Returns a JSON string representation from this UpdatesLogEntry object
+   */
+  public func asString() -> String? {
+    do {
+      let jsonEncoder = JSONEncoder()
+      let jsonData = try jsonEncoder.encode(self)
+      return String(data: jsonData, encoding: .utf8)
+    } catch {
+      return nil
+    }
+  }
+
+  /**
+   Returns a Dictionary representation from this UpdatesLogEntry object
+   */
+  public func asDict() -> [String: Any] {
+    var result = [String: Any]()
+    result["timestamp"] = timestamp
+    result["message"] = message
+    result["code"] = code
+    result["level"] = level
+    if (updateId != nil) {
+      result["updateId"] = updateId
+    }
+    if (assetId != nil) {
+      result["assetId"] = assetId
+    }
```

Comment:
Generally, we don't use `()` in ifs, switches, etc.
Is there any reason why you check if it's non-nil? I think you can just assign it and don't care 😅

Reply:
I did this to ensure that OC code got the dictionaries with the right behavior. I'll check to see if that is still necessary.

Reply:
Since this is Swift dictionary, assigning `nil` means nothing else than just deleting the key from the dictionary if it existed

Reply:
Fixed the syntax